app/v3/controllers/RetrieveForeignPropertyBsasController.scala

```diff
@@ -65,11 +65,9 @@ class RetrieveForeignPropertyBsasController @Inject()(
         for {
           parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawData))
           response      <- EitherT(service.retrieveForeignPropertyBsas(parsedRequest))
-          hateoasResponse <- EitherT.fromEither[Future](
-            hateoasFactory
-              .wrap(response.responseData, RetrieveForeignPropertyHateoasData(nino, calculationId))
-              .asRight[ErrorWrapper])
         } yield {
+          val hateoasResponse = hateoasFactory.wrap(response.responseData, RetrieveForeignPropertyHateoasData(nino, calculationId, None))
```

Comment:
Maybe the following to keep it consistent ?
          val hateoasData     =  ..
          val hateoasResponse =  ..